ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/regression/LinearRegression.java

```diff
@@ -208,16 +223,29 @@ public MLOutput predict(DataFrame dataFrame, Model model) {
     }
 
     @Override
-    public Model train(DataFrame dataFrame) {
+    public MLOutput predict(MLInputDataset inputDataset, MLModel model) {
+        if (model == null) {
+            throw new IllegalArgumentException("No model found for linear regression prediction.");
+        }
+
+        regressionModel = (org.tribuo.Model<Regressor>) ModelSerDeSer.deserialize(model);
```

Comment:
how will we reach here? also what's the usage of `regressionModel`?

Reply:
Not quite get your question. You mean how we deserialize the model?
Regression model here is just for linear regression.

Reply:
Sorry my bad,  I misinterpret the code.